src/styles/block/calendar.scss

```diff
@@ -0,0 +1,62 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+
+  gap: $gap;
+  padding: $paddings;
+  width: $boxSize * 7 + $gap *6 + 20px;
+
+  &__days {
+    box-sizing: border-box;
+
+    width: $boxSize;
+    height: $boxSize;
+
+    background-color: #eee;
+    border: 1px solid black;
+    transition: transform 0.8s;
+
+    cursor: pointer;
+
+    &:hover {
+      transform: translateY(-20px);
+      background-color: purple;
+    }
+
+    @for $i from 28 to 30 {
+      .calendar__length--#{$i} > {
+        .calendar__days:not(:nth-child(-n + #{$i})) {
```

Comment:
try check your selector without cycle, when it works add cycle
try without :not with n + index